### ISSUE TITLE
Heading levels incorrect for accordion web component

### ISSUE THREAD

**Issue by adamwhitlock1** [department-of-veterans-affairs/va.gov-team#30359]
## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh or Adam

## User Story or Problem Statement

As a screen reader user, I expect heading levels to be consistent with the surrounding content when accessing accordions.

## Details

The original issue that brought this to light is here:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/27619

There needs to be a way to ensure that accordion headings maintain the correct hierarchy level in relation to their surrounding content. 
![heading level mismatch example](https://user-images.githubusercontent.com/14154792/127920821-22d4d1e7-d86f-40bc-a5c9-8bc4435d330b.png)

## Acceptance Criteria

- [ ] Accordions use the correct heading level

## Steps to Recreate

1. View accordions on https://www.va.gov/health-care/covid-19-vaccine/
2. Inspect accordion heading levels

## Proposed Solution

Detect the previous heading level in the document and set accordion headings based on this level.

**Comment by laflannery** [comment]
This component is also used on the VAMC Top Task Medical Records Office template but the headings structure is different:
  * Example: https://www.va.gov/palo-alto-health-care/medical-records-office/
  * There is a heading for "Questions about medical records" which is an H2
  * Below are the Q&A accordions which are also H2 - they should be H3 in order to follow logical heading structure

We may need to determine first where this component is being used in order to ensure these headings fit into the page structure whenever possible

**Comment by wesrowe** [comment]
@laflannery, thanks for the example link. Did you also explore the Covid example (screenshot above)? Key info is missing from the screenshot, and I want to understand the breadth of the problem before we pull in many more people. (Or ask Dave about prioritization.)

**Comment by laflannery** [comment]
Sorry to be unclear @wesrowe - I did look at the above example.  In that case the accordions are still H2 but the section heading above is an H3 so in this case the accordions should be H4.

It would seem that the accordions are always being added as H2 regardless of the heading structure on the page so would it be possible to change the heading depending on where the component is used?

**Comment by jilladams** [comment]
In discussion with Dave today, he flagged potentially a newer accordion issue from the last few months where the design component was updated with trickle down mitigation issues. We should verify if this ticket is valid in the modern state of the world, or has new nuances. @laflannery first step, are you able to verify that you see things behaving in the way described here today?

**Comment by laflannery** [comment]
@jilladams I'm not sure I understand the request? However, I can confirm the 2 pages referenced (the COVID page and the medical records page) both currently have these heading issues on the live site now

I also just now reviewed the [Component ](https://design.va.gov/components/accordion) in the Design System and was able to find an [issue ](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/727) related to headings which suggests the component does have the ability to be wrapped in other header tags

**Comment by jilladams** [comment]
Ok. Per discussions today with @wesrowe and Dave, we'll own this ticket. Before we can fix, we'll need a discovery ticket to understand the mechanisms and make sure we're familiar with the issue / design system component. Wes to file. Moving this to `va.gov-cms` to track in our board.
